metadata-ingestion/src/datahub/ingestion/source/datahub/datahub_kafka_reader.py

```diff
@@ -27,24 +28,26 @@ class DataHubKafkaReader(Closeable):
     def __init__(
         self,
         config: DataHubSourceConfig,
+        connection_config: KafkaConsumerConnectionConfig,
```

Comment:
Passed in separately so that it can be non-optional, avoiding assertions

Reply:
What if we only passed in one thing, but then did this in init

```python
assert self.config.kafka_connection
self.connection_config = self.config.kafka_connection
```

Reply:
Same thing here, in general I don't like assertions -- this requires callers to pass in a non-optional consumer connection, rather than requiring them to know that the class assumes it exists